File: docs/source/src/python/user-guide/expressions/searching.py
# --8<-- [start:dfnum]
import polars as pl

df = pl.DataFrame(
    {
        "integers": [17, 42, None, 35],
    }
)

print(df)
# --8<-- [end:dfnum]

# --8<-- [start:index_of]
result = df.select(pl.col("integers").index_of(35))
print("First index of 35 is:", result)
result = df.select(pl.col("integers").index_of(None))
print("First index of null is:", result)
# --8<-- [end:index_of]

# --8<-- [start:index_of_not_found]
result = df.select(pl.col("integers").index_of(1233))
print("Index of unfound number is:", result)
# --8<-- [end:index_of_not_found]

File: docs/source/src/rust/user-guide/expressions/searching.rs
fn main() -> Result<(), Box<dyn std::error::Error>> {
    // --8<-- [start:dfnum]
    use polars::prelude::*;

    let df = df! (
        "integers" => &[Some(17), Some(42), None, Some(35)],
    )?;
    // --8<-- [end:dfnum]

    // --8<-- [start:index_of]
    let result = df
        .clone()
        .lazy()
        .select([col("integers").index_of(35)])
        .collect()?;
    println!("{}", result);

    let result = df
        .clone()
        .lazy()
        .select([col("integers").index_of(Null {}.lit())])
        .collect()?;
    println!("{}", result);
    // --8<-- [end:index_of]

    // --8<-- [start:index_of_not_found]
    let result = df
        .clone()
        .lazy()
        .select([col("integers").index_of(1233)])
        .collect()?;
    println!("{}", result);
    // --8<-- [end:index_of_not_found]

    Ok(())
}
